core/src/test/java/org/elasticsearch/common/xcontent/ConstructingObjectParserTests.java

```diff
@@ -172,7 +220,7 @@ void setFoo(String foo) {
         ConstructingObjectParser<CalledOneTime, ParseFieldMatcherSupplier> parser = new ConstructingObjectParser<>("one_time_test",
                 (a) -> new CalledOneTime((String) a[0]));
         parser.declareString(CalledOneTime::setFoo, new ParseField("foo"));
-        parser.declareString(constructorArg(), new ParseField("yeah"));
+        parser.declareString(randomBoolean() ? constructorArg() : optionalConstructorArg(), new ParseField("yeah"));
```

Comment:
Can we expand this test case to include parsing some JSON that does not include the `yeah` field *when* the constructor argument is optional?